shared/src/commonMain/kotlin/co/nimblehq/blisskmmic/data/network/datasource/NetworkDataSource.kt

```diff
@@ -4,22 +4,32 @@ import co.nimblehq.blisskmmic.data.model.ResetPasswordMeta
 import co.nimblehq.blisskmmic.data.network.core.NetworkClient
 import co.nimblehq.blisskmmic.data.network.target.LoginTargetType
 import co.nimblehq.blisskmmic.data.network.target.ResetPasswordTargetType
+import co.nimblehq.blisskmmic.data.model.PaginationMetaApiModel
+import co.nimblehq.blisskmmic.data.model.SurveyApiModel
+import co.nimblehq.blisskmmic.data.network.helpers.requestBuilder
+import co.nimblehq.blisskmmic.data.network.target.SurveySelectionTargetType
 import co.nimblehq.blisskmmic.domain.model.TokenApiModel
 import kotlinx.coroutines.flow.Flow
 
 interface NetworkDataSource {
 
     fun logIn(target: LoginTargetType): Flow<TokenApiModel>
     fun resetPassword(target: ResetPasswordTargetType): Flow<ResetPasswordMeta>
+    fun survey(target: SurveySelectionTargetType): Flow<Pair<List<SurveyApiModel>, PaginationMetaApiModel>>
 }
 
 class NetworkDataSourceImpl(private val networkClient: NetworkClient): NetworkDataSource {
 
     override fun logIn(target: LoginTargetType): Flow<TokenApiModel> {
-        return networkClient.fetch(target.requestBuilder)
+        return networkClient.fetch(target.requestBuilder())
     }
 
     override fun resetPassword(target: ResetPasswordTargetType): Flow<ResetPasswordMeta> {
-        return networkClient.fetch(target.requestBuilder)
+        return networkClient.fetch(target.requestBuilder())
+    }
+
+    override fun survey(target: SurveySelectionTargetType):
+            Flow<Pair<List<SurveyApiModel>, PaginationMetaApiModel>> {
+        return networkClient.fetchWithMeta(target.requestBuilder())
```

Comment:
The indentation here looks a bit too much, do you mind checking it again?
```suggestion
    override fun survey(target: SurveySelectionTargetType):
        Flow<Pair<List<SurveyApiModel>, PaginationMetaApiModel>> {
        return networkClient.fetchWithMeta(target.requestBuilder())
```

FYI, this should be the standard indentations setting for AS 😁
![Screenshot 2022-12-09 at 16 23 58](https://user-images.githubusercontent.com/70877098/206669034-325d1109-8a46-40ee-8272-6693d5f2296d.png)